### ISSUE TITLE
[Bug]: validateDOMNesting console error in TableContainer description

### ISSUE THREAD

**Issue by katsoohoo** [carbon-design-system/carbon#14163]
### Package

carbon-components-react

### Browser

Chrome

### Package version

v7.51.0

### React version

v16.13.1

### Description

When using a `Tooltip` inside the `description` of a `TableContainer`, the follow error is logged in the console: `Warning: validateDOMNesting(...): <div> cannot appear as a descendant of <p>.`

Ideally, would like a similar solution to https://github.com/carbon-design-system/carbon/issues/3901

### Reproduction/example

https://codesandbox.io/s/trusting-pike-mrtm4c

### Steps to reproduce

1. Create a `DataTable`
2. Add a `Tooltip` in the `TableContainer`'s `description`
3. In the console, the following error is logged: `Warning: validateDOMNesting(...): <div> cannot appear as a descendant of <p>.`

Paste code in [CodeSandbox template for carbon-components-react](https://codesandbox.io/s/github/carbon-design-system/carbon/tree/v10/packages/react/examples/codesandbox) to recreate.
```tsx
import React from "react";
import { render } from "react-dom";
import { DataTable, TableContainer, Tooltip } from "carbon-components-react";

const App = () => (
  <div>
    <DataTable rows={[]} headers={[]}>
      {({ rows, headers }) => (
        <TableContainer
          title={"title"}
          description={
            <React.Fragment>
              my description
              <Tooltip>{"my tooltip"}</Tooltip>
            </React.Fragment>
          }
        />
      )}
    </DataTable>
  </div>
);

render(<App />, document.getElementById("root"));
```

### Suggested Severity

Severity 4 = Unrelated to a user task, has a workaround or does not need a workaround.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Yeah, this could probably be swapped to a `div` without any severe repercussions

**Comment by jose-biescas** [comment]
Can I get assigned this and fix it? Thank you
